Installation.md

```diff
@@ -0,0 +1,267 @@
+```
+$ sudo add-apt-repository ppa:webupd8team/java
+$ sudo apt-get update
+$ sudo apt-get install oracle-java$JAVA_VERSION-installer
+```
+
+##### Install dependencies #####
+
+TMC-server dependencies
+```
+$ sudo apt-get install git build-essential zip unzip imagemagick maven make bc xvfb postgresql postgresql-contrib
+$ sudo apt-get install chrpath libssl-dev libxft-dev \
+  libfreetype6 libfreetype6-dev libfontconfig1 libfontconfig1-dev
+```
+Ruby dependencies
+```
+$ sudo apt-get install git-core curl zlib1g-dev libssl-dev libreadline-dev libyaml-dev libsqlite3-dev sqlite3 libxml2-dev libxslt1-dev libcurl4-openssl-dev python-software-properties
+```
+RVM dependencies
+```
+$ sudo apt-get install libgdbm-dev libncurses5-dev automake libtool bison libffi-dev
+```
+TMC-sandbox dependencies
+```
+$ sudo apt-get install squashfs-tools multistrap e2fsprogs e2tools
+```
+
+##### Installing PhantomJS #####
+
+Download PhantomJS from their [website](http://phantomjs.org/)
+```
+$ wget https://bitbucket.org/ariya/phantomjs/downloads/$PHANTOM_JS.tar.bz2
+```
+After download finishes extract the files and create symlinks
+```
+$ tar -xvjf $PHANTOM_JS.tar.bz2
+$ sudo mv $PHANTOM_JS /usr/local/share
+$ sudo ln -s /usr/local/share/$PHANTOM_JS/bin/phantomjs /usr/local/bin
+$ rm $PHANTOM_JS.tar.bz2
+```
+Check that PhantomJS is installed correctly
+```
+$ phantomjs --version
+```
+
+### Install ruby via RVM ###
+#### Install RVM as multi-user install ####
+
+```
+$ gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+$ \curl -L https://get.rvm.io | sudo bash -s stable
+```
+
+After installation finishes add new user and setup groups
+
+```
+$ sudo adduser tmc
+$ sudo addgroup rvm
+$ sudo usermod -a -G rvm tmc
+$ sudo /usr/local/rvm/bin/rvm install $RUBY_VERSION
+$ sudo /usr/local/rvm/bin/rvm use $RUBY_VERSION --default
+```
+
+Setup ruby path
+
+```
+$ echo 'export PATH=$PATH:/usr/local/rvm/rubies/ruby-$RUBY_VERSION/bin' | sudo tee -a etc/profile.d/ruby.sh
+```
+
+#### Switch to tmc user and login their shell ####
+
+```
+$ sudo su tmc
+$ bash -l
+$ cd
+```
+
+#### Import public key for rvm installation ####
+
+```
+$ command curl -sSL https://rvm.io/mpapis.asc | gpg --import -
+$ rvm get stable
+```
+
+#### Setup user environment variables ####
+
+```
+$ echo "export GEM_HOME=$(ruby -e 'print Gem.user_dir')" >> ~/.bashrc
+$ echo "export PATH=$(echo $PATH):$GEM_HOME/bin" >> ~/.bashrc
+$ export GEM_HOME=$(ruby -e 'print Gem.user_dir')
+$ export PATH=$PATH:$GEM_HOME/bin
+```
+
+### PostgreSQL ###
+#### Create postgres user ####
+
+Run following command and when prompted give **tmc** as a password, if you want to use another username or stronger password change them into **config/database.yml**.
+
+```
+$ createuser tmc -s -P
+```
+
+If you cannot run above command, you can also create the user manually
+
+```
+$ sudo -s
+$ su - postgres
+$ psql
+    CREATE USER tmc WITH SUPERUSER PASSWORD 'tmc';
+```
```

Comment:
Would this single command work?
`sudo -u postgres psql -c "CREATE USER tmc WITH SUPERUSER PASSWORD 'tmc';"`